lib/http-proxy/passes/web-outgoing.js

```diff
@@ -80,12 +80,44 @@ var redirectRegex = /^30(1|2|7|8)$/;
    *
    * @api private
    */
-  function writeHeaders(req, res, proxyRes) {
+  function writeHeaders(req, res, proxyRes, options) {
+    var rewriteCookieDomainConfig = options ? options.cookieDomainRewrite : undefined;
+    if(typeof rewriteCookieDomainConfig === "string") { //also test for ""
+      rewriteCookieDomainConfig = {"*": rewriteCookieDomainConfig};
+    }
     Object.keys(proxyRes.headers).forEach(function(key) {
-      if(proxyRes.headers[key] != undefined){
-        res.setHeader(String(key).trim(), proxyRes.headers[key]);
+      var header = proxyRes.headers[key];
+      if(header != undefined){
+        if(rewriteCookieDomainConfig && key.toLowerCase() === 'set-cookie') {
+          header = rewriteCookieDomain(header);
+        }
+        res.setHeader(String(key).trim(), header);
       }
     });
+
+    function rewriteCookieDomain(header) {
```

Comment:
this function should move to the `common.js` file as it is not a `pass` that accepts the `(req, res, ...)` signature.